src/Compilers/CSharp/Test/Semantic/Semantics/SwitchTests.cs

```diff
@@ -1155,6 +1155,25 @@ public static void Test(E x)
             Assert.NotNull(symbolInfo);
         }
 
+        [WorkItem(31587, "https://github.com/dotnet/roslyn/issues/31587")]
+        [Fact]
+        public void NullableDecimalToEnumConversion()
+        {
+            var source = @"
+using System;
+enum e1 { v1, v2 }
+public class C {
+    public void M(Decimal? d) {
+        switch ((e1)d) {
+            case e1.v1: break;
+            default: break;
+        }
+    }
+}
+";
+            CompileAndVerify(source).VerifyDiagnostics();
+        }
```

Comment:
It looks like the repro does not depend on `switch` since the following hits the issue:
```C#
enum E { }
class C
{
    static E F(decimal? d) => (E)d;
}
```
Consider simplifying the test and moving to a conversions test class, perhaps `CodeGenConversionTests`.

Reply:
Sounds good.